isis/src/docsys/documents/DeveloperAddFeature/DeveloperAddFeature.xml

```diff
@@ -0,0 +1,218 @@
+            </p>
+            <p>
+              Depending on how your computer is setup, you may also need to install:
+            </p>
+            <ul>
+              <li> Compiler (Mac OS Xcode, Linux GNU C++)</li>
+              <li> Editor (Vim, Nano, Atom, ...</li>
+            </ul>
+
+            <p>
+              Once you have a complete environment set up for ISIS development, we suggest you 
+              <a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#building-isis3">compile and build</a> 
+              all of ISIS. This step will make sure sure you have everything setup and ready to go.
+            </p>
+
+          <h2><a name="Starting a conversation with other ISIS developers">Starting a conversation with other ISIS developers</a></h2>
+            <p>
+              Before you start writing your new feature, create an <a href="https://github.com/USGS-Astrogeology/ISIS3">issue</a> for 
+              it on the ISIS GitHub site. Doing this will allow other ISIS developers to way in on your preposed feature. By posting
+              a description of what you want to do and asking questions, other developers can give you pointers on how to 
+              implement it. Label the issue as an "enhancement", to give others an idea what kind of issue it is.
+            </p>
+
+          <h2><a name="Writing your new application">Writing your new application</a></h2>
+
+            <h3>Creating the files for your new application</h3>
+            <p>
+              The ISIS source code is organized by having a dedicated directory where all of the source code for an application is
+              stored. The files in this directory will be exclusively to your feature. This is where using another application as
+              a template can be very useful.
+            </p>
+            <p>
+              The source code inside an ISIS application directory should be organized to be a callable ISIS 
+              application. This is a change from the older style applications that could not be called directly by other applications.
+              Details about writing callable ISIS applications can be found at: 
+              <a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Writing-ISIS3-Tests-Using-Gtest-and-Ctest#refactoring-isis3-applications">Refactoring ISIS Applications</a>
+            </p>
+
+            <ol>
+                <li>Decide where your new application belongs</li>
+                <li>Create a directory using the same name as you new application</li>
+                <li>Create a "main.cpp" file</li>
+                <li>Create a "your_new_app.cpp" file</li>
+                <li>Create a "your_new_app.h" file</li>
+                <li>Create a "your_new_app.xml" file</li>
+                <li>Create any additional support files (e.g., Support_class.cpp, Support_class.h</li>
```

Comment:
Update filenames to match style guide and coding standards:

yournewapp.cpp
SupportClass.cpp
...